projects/lua/build.sh

```diff
@@ -0,0 +1,221 @@
+#!/bin/bash -eu
+# Copyright 2023 Sergey Bronnikov
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+CC=clang
+CXX=clang++
+CFLAGS="-g -fsanitize=fuzzer-no-link,address,integer,bounds,null,undefined,float-divide-by-zero"
+CXXFLAGS="-g -fsanitize=fuzzer-no-link,address,integer,bounds,null,undefined,float-divide-by-zero"
+
+cd /testdir
+
+: ${LD:="${CXX}"}
+: ${LDFLAGS:="${CXXFLAGS}"}  # to make sure we link with sanitizer runtime
+
+cmake_args=(
+    -DUSE_LUA=ON
+    -DOSS_FUZZ=OFF
+    -DENABLE_ASAN=ON
+    -DENABLE_UBSAN=ON
+    -DCMAKE_BUILD_TYPE=Debug
+
+    # C compiler
+    -DCMAKE_C_COMPILER="${CC}"
+    -DCMAKE_C_FLAGS="${CFLAGS}"
+
+    # C++ compiler
+    -DCMAKE_CXX_COMPILER="${CXX}"
+    -DCMAKE_CXX_FLAGS="${CXXFLAGS}"
+
+    # Linker
+    -DCMAKE_LINKER="${LD}"
+    -DCMAKE_EXE_LINKER_FLAGS="${LDFLAGS}"
+    -DCMAKE_MODULE_LINKER_FLAGS="${LDFLAGS}"
+    -DCMAKE_SHARED_LINKER_FLAGS="${LDFLAGS}"
+)
+
+# Build fuzzers.
+[[ -e build ]] && rm -rf build
+cmake "${cmake_args[@]}" -S . -B build -G Ninja
+cmake --build build --parallel
+
+# Archive and copy to $OUT seed corpus if the build succeeded.
+for f in $(find build/tests/ -name '*_test' -type f);
+do
+  name=$(basename $f);
+  module=$(echo $name | sed 's/_test//')
+  corpus_dir="test/static/corpus/$module"
+  echo "Copying for $module";
+  cp $f /
+  [[ -e $corpus_dir ]] && cp -r $corpus_dir /corpus_$module
```

Comment:
None corpus directory was created after launching this script. Maybe the issue is about paths?

Reply:
The problem was in the absence of corpus itself :)
Now the corpus is created [^1] and used in the script.

[^1]: https://github.com/ligurio/lua-c-api-corpus

```diff
--- a/projects/lua/Dockerfile
+++ b/projects/lua/Dockerfile
@@ -30,6 +30,9 @@ RUN git clone https://github.com/ligurio/lua-c-api-tests testdir
 
 WORKDIR testdir
 
+# Clone corpus from GitHub.
+RUN git clone https://github.com/ligurio/lua-c-api-corpus corpus
+
 # Checkout specified commit. It could be updated later.
 RUN git checkout 017105bb03982f558e5111748bdfd44759699a79
 
```